tenders/common/mft.c

```diff
@@ -65,28 +65,37 @@ int mft_validate(struct mft *mft, size_t mft_size)
 
     for (unsigned i = 0; i != mft->entries; i++) {
         /*
-         * Ensure name[] is always terminated with \0, even if there is garbage
+         * Ensure .name[] is always terminated with \0, even if there is garbage
          * in the array itself.
          */
-        mft->e[i].name[MFT_NAME_MAX] = 0;
+        if (mft->e[i].name[MFT_NAME_MAX] != 0)
+            return -1;
         /*
          * Sanitize private fields (to be used by the tender/bindings):
          *
-         * 1. b.hostfd must be set to an invalid value, attached to false.
+         * 1. .attached must be false (which implies that .u is uninitialised)
```

Comment:
I guess it also implies that `mft->e[i].b.hostfd` is uninitialized?

Reply:
Yeah, all of `.b`, will add to comment. I had to remove setting `.b.hostfd = -1` to preserve constness.